train.lua

```diff
@@ -131,6 +147,71 @@ local function loadModel(opt, dicts)
 
   checkpoint, opt, paramChanges = onmt.train.Saver.loadCheckpoint(opt)
 
+  if checkpoint.dicts ~= dicts and opt.update_vocab then
+    _G.logger:info(' * Found different vocabularies in the new train-set')
```

Comment:
Could this whole block code (under the `if opt.update_vocab then` condition) be in a separate function?